chutney/server/src/test/java/com/chutneytesting/execution/domain/campaign/CampaignExecutionEngineTest.java

```diff
@@ -108,37 +111,31 @@ public static void setUpAll() {
     @BeforeEach
     public void setUp() {
         sut = new CampaignExecutionEngine(campaignRepository, campaignExecutionRepository, scenarioExecutionEngine, scenarioExecutionEngineAsync, executionHistoryRepository, testCaseRepository, jiraXrayPlugin, metrics, executorService, datasetRepository, objectMapper);
-        firstTestCase = createGwtTestCase("1");
-        secondTestCase = createGwtTestCase("2");
-        when(testCaseRepository.findExecutableById(firstTestCase.id())).thenReturn(of(firstTestCase));
-        when(testCaseRepository.findExecutableById(secondTestCase.id())).thenReturn(of(secondTestCase));
-        when(executionHistoryRepository.getExecution(eq(firstTestCase.id()), or(eq(0L), eq(10L))))
-            .thenReturn(executionWithId(firstTestCase.id(), firstScenarioExecutionId));
-        when(executionHistoryRepository.getExecution(eq(secondTestCase.id()), or(eq(0L), eq(20L))))
-            .thenReturn(executionWithId(secondTestCase.id(), secondScenarioExecutionId));
+        firstTestCase = createAndMockExecutedGwtTestCase(firstScenarioExecutionId, SUCCESS);
+        secondTestCase = createAndMockExecutedGwtTestCase(secondScenarioExecutionId, SUCCESS);
+
         when(scenarioExecutionEngine.execute(any(ExecutionRequest.class))).thenReturn(mock(ScenarioExecutionReport.class));
         when(scenarioExecutionEngine.saveNotExecutedScenarioExecution(any(ExecutionRequest.class))).thenReturn(mock(ExecutionHistory.Execution.class));
     }
 
     @Test
     public void should_update_jira_xray() {
```

Comment:
Maybe it would be good to change this rule by `should_update_jira_xray_for_completed_execution` or something like that ?